Python/traceback.c

```diff
@@ -1101,3 +1104,59 @@ _Py_DumpTracebackThreads(int fd, PyInterpreterState *interp,
     return NULL;
 }
 
+/* This is for faulthandler.
+ * Apparently, backtrace() doesn't play well across DLL boundaries on macOS */
+#if defined(HAVE_EXECINFO_H) && defined(HAVE_BACKTRACE) && defined(HAVE_BACKTRACE_SYMBOLS)
+void
+_Py_DumpStack(int fd)
+{
+#define BACKTRACE_SIZE 32
+#define TRACEBACK_ENTRY_MAX_SIZE 256
+    PUTS(fd, "Current thread's C stack trace (most recent call first):\n");
+    void *callstack[BACKTRACE_SIZE];
+    int frames = backtrace(callstack, BACKTRACE_SIZE);
+    if (frames == 0) {
+        // Some systems won't return anything for the stack trace
+        PUTS(fd, "  <system returned no stack trace>\n");
+        return;
+    }
+
+    char **strings = backtrace_symbols(callstack, BACKTRACE_SIZE);
+    if (strings == NULL) {
+        PUTS(fd, "  <not enough memory to get stack trace>\n");
+        return;
+    }
+    for (int i = 0; i < frames; ++i) {
+        char entry_str[TRACEBACK_ENTRY_MAX_SIZE];
+        snprintf(entry_str, TRACEBACK_ENTRY_MAX_SIZE, "  %s\n", strings[i]);
+        size_t length = strlen(entry_str) + 1;
+        if (length == TRACEBACK_ENTRY_MAX_SIZE) {
```

Comment:
We could extract this in a function as this could perhaps improve readability although you're not forced to